src/async-split-at.mjs

```diff
@@ -0,0 +1,69 @@
+import ensureAsyncIterable from './internal/ensure-async-iterable'
+import curry from './internal/curry'
+
+function asyncSplitAt (index, iterable) {
+  const iter = ensureAsyncIterable(iterable)[Symbol.asyncIterator]()
+  let position = 0
+  let firstCache = null
+  let firstCacheStart = null
+  let done = false
+
+  async function cleanUp () {
+    if (!done && typeof iter.return === 'function') {
+      await iter.return()
+    }
+  }
+
+  async function * first () {
+    try {
+      while (!done) {
+        if (firstCache) {
+          yield firstCache[position - firstCacheStart]
```

Comment:
This line should be `yield await firstCache[...]`

Reply:
Just think about it, it makes sense. The `await` needs to go in the loop because e still want `first` to be able to get each item as soon as it is ready.  And it's the most direct translation of what the logic was: whatever the iterator actually yielded, that's what belongs in the cache.